Fix bug with missing 'Call reception message'

neilbmclaughlin opened the pull request:
Fixes #60

See issue for example search.

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

neilbmclaughlin commented:
Fixes #63 

See issue for example search.

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
+          done();
+        });
+    });
+    describe('matching surgeries found', () => {
```
Extra line please

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
+            });
+        });
+        it('should display singular message text', (done) => {
+          chai.request(app)
```
Can this request and the one above be incorporated into one and moved into the `describe` block so the validation of the contents of cheerio can be tested within the `it` blocks. This should reduce duplication and ease readability.

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
-        expect($('.link-back:first-of-type').eq(1).text()).to.equal('Back');
-        expect($('.link-back').attr('href')).to.equal(`${constants.SITE_ROOT}`);
-        done();
+      describe('multiple matches', () => {
```
Extra line please

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
+        });
+        it('should display plural message text', (done) => {
+          const search = 'Surgery';
+          chai.request(app)
```
Same as previous comment about removing duplication and running the request within the `describe` block

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
+          });
+      });
+    });
+    describe('when the surgery does not have a website', () => {
```
Extra line please

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -169,8 +248,8 @@ describe('Results page error handling', () => {
     });
   });
   describe('when search is an empty string', () => {
```
Extra line please

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -186,31 +265,22 @@ describe('Results page error handling', () => {
   });
-
-  describe("when search doesn't bring back results", () => {
+  describe('when search is some empty spaces', () => {
```
Extra line please

st3v3nhunt commented on test/integration/resultsPage.js:
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
+          });
       });
+    });
+    describe('Surgeries without phone number', () => {
```
Extra line please

neilbmclaughlin commented:
@st3v3nhunt - will refactor the tests to incorporate your suggestions.

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

neilbmclaughlin commented:
@st3v3nhunt - I have addressed some of the issues you raise but would like to address the issue of refactoring the tests to use Promises, beforeEach and describe block better as part of a separate pull request so the bug fix here can be released. The reason being was that I was getting bogged down with issues with Docker volumes. Raised issue #66 for the specific problem of too much logic in the views.

Are you OK the merge on the basis of this?

neilbmclaughlin commented on test/integration/resultsPageErrors.js:
```diff
@@ -0,0 +1,80 @@
+
+const resultsRoute = `${constants.SITE_ROOT}/results/`;
+
+function assertSearchResponse(search, assertions) {
```
I realise there is duplication here with the resultsPage tests but it is just an interim step before we use beforeEach.

st3v3nhunt commented:
@neilbmclaughlin There is still a bunch of inconsistent spacing within the tests either no new line where there should be or an additional new line where there shouldn't be. Will this also be dealt with in the next change? Ideally it would be good to get it done as part of this

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

neilbmclaughlin commented:
@st3v3nhunt - experimented with a test specific lint rule around padding. This may not be quite what you want but hopefully is close enough. Importantly it allows us to pick issues up automatically rather than at the review stage and allows us to fix them automatically using `npm run lint -- --fix`. The is perhaps a discussion to be had as to whether the test specific padding rule should be applied to the whole of the code base since `--fix` makes it a trivial task.

st3v3nhunt commented:
@neilbmclaughlin I think that change makes it harder to read. The previous set of comments about where to add lines and remove them covered all of the suggestions.
Which equate to this...the general rule I work with, and is implicitly used by other people across the set of `connecting-to-service` associated repos is have no padding for blocks with the exception of between `it` blocks. The reasoning for padding between `it` blocks is to aid readability and have each individual test be visibly separate from one another.
Looking at the eslint rule for [padded-blocks](http://eslint.org/docs/rules/padded-blocks) there isn't the configurability to specify a rule based on mocha flavoured blocks.

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

neilbmclaughlin commented:
@st3v3nhunt - I've overridden the no-padding rule so we can pad as we wish in tests and `git push --force`'d a reformatted set of tests. This will allow us to visually separate test blocks without adding unnecessary white space.  My frustration with this was that the repo eslint rule seemed to be at odds with what you think we should be doing - rules have make coherent sense to me in order to want to do them. I think if we want to preceded a test block with a blank line (and I think this is a good idea) it should apply regardless of how many tests there are and whether it is an 'it' or a 'describe'. Hopefully we are converging :)

c2s-dev commented:
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)
